### DISCUSSION TITLE
L'ordre des questions est changé si la question est répondu ou non

### DISCUSSION THREAD

**Discussion by wiinxt** [betagouv/mon-entreprise#2714] [Idées]
Si on répond à une question, qu'on clique sur précédent, puis sur suivant alors on sera plus sur la question qu'on avait répondu.

https://user-images.githubusercontent.com/6691954/154672917-7bb877de-261a-4c80-83b7-e9ec8668bc3a.mp4

**Comment by johangirod** [comment]
Effectivement, c'est un problème dû à l'impossibilité de savoir la différence entre une règle rendue non applicable ou une règle dont la valeur dans la situation est `non`, mais qui est applicable.

Par exemple, si l'on répond "non" à la question "connaissait vous votre taux AT/MAP", puis "non" à "l'activité est-elle sans aucun risque", que l'on revient en arrière et que l'on répond "oui" pour le taux, il n'y a aucun moyen de savoir si la question sur le risque est "non" parce que c'est dans la situation ou parce qu'elle est non applicable. 

On préfère donc ne pas afficher les questions déjà répondues. 

Cela va changer avec [la prochaine version de publicodes](https://github.com/betagouv/publicodes/pull/159) : on pourra différencier une règle non applicable (null) d'une règle qui vaut "non".

**Comment by johangirod** [comment]
Après discussion avec @wiinxt, nous en avons conclu qu'il faudrait tout de même améliorer l'actuel. On pourrait sauvegarder la pile des questions passées via le bouton précédent et la restaurer lors du clique sur "suivant", tout en filtrant les questions devenues non applicables (via `UNSAFE_isNotApplicable` pour le moment).

**Comment by johangirod** [comment]
Retour utilisateur de @leo-terrier dans #2751 qui remonte le même problème : 

> J'ai remarqué que lorsqu'une question n'est posée (sous le simulateur) que lorsqu'une réponse antérieure est positive, après plusieurs modification de cette réponse antérieure, la question n'est plus accessible dans le questionnaire sous le simulateur, mais uniquement via la modification globale des réponses.

### Reproduction du problème

**Simulation EURL**
```
Activité libérale : oui
Clicker suivant

Activité réglementée : oui
Clicker suivant

Retour en arrière
activité libérale : non, 
clicker suivant

retour en arrière
activité libérale : oui 
clicker suivant
```
La question "activité réglementée" a disparu

**Comment by liliced** [comment]
Obsolète avec #3136
